.github/workflows/code.yml

```diff
@@ -0,0 +1,213 @@
+        lock: [false]
+
+        include:
+          # Add an extra Python3.10 runner to use the lockfile
+          - os: "ubuntu-latest"
+            python: "3.10"
+            lock: true
+
+    runs-on: ${{ matrix.os }}
+    env:
+      # https://github.com/pytest-dev/pytest/issues/2042
+      PY_IGNORE_IMPORTMISMATCH: "1"
+      # enable QT tests with no X Display
+      QT_QPA_PLATFORM: "offscreen"
+
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+
+      - name: setup python ${{ matrix.python }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python }}
+
+      - name: create requirements.txt
+        if: ${{ matrix.lock == true }}
+        run: |
+          touch requirements.txt
+          pip install -r requirements.txt -e .
+          pip freeze --exclude-editable > requirements.txt
+
+      - name: run tests
+        run: |
+          if [ '${{ matrix.lock == true }}' ]; then lock='-r requirements_dev.txt'; fi
+          touch requirements_dev.txt
+          pip install ${lock} -e .[dev]
+          pytest tests
+
+      - name: create requirements_dev.txt
+        run: |
+          pip freeze --exclude-editable > requirements_dev.txt
+
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v2
+        with:
+          name: ${{ matrix.python }}/${{ matrix.os }}/${{ matrix.lock }}
+          files: cov.xml
+
+      - name: Upload lockfiles
+        if: ${{ matrix.lock == true }}
+        uses: actions/upload-artifact@v2
+        with:
+          name: lockfiles
+          path: |
+            requirements.txt
+            requirements_dev.txt
+
+  release:
+    needs: [lint, wheel, test]
+    runs-on: ubuntu-latest
+    # upload to PyPI and make a release on every tag
+    if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
+    steps:
+      - uses: actions/download-artifact@v2
+        with:
+          path: artifacts
+
+      - name: Github Release
+        # We pin to the SHA, not the tag, for security reasons.
+        # https://docs.github.com/en/actions/learn-github-actions/security-hardening-for-github-actions#using-third-party-actions
+        uses: softprops/action-gh-release@1e07f4398721186383de40550babbdf2b84acfc5 # v0.1.14
+        with:
+          files: |
+            artifacts/dist/*
+            artifacts/lockfiles/*
+          generate_release_notes: true
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Publish to PyPI
+        env:
+          TWINE_USERNAME: __token__
+          TWINE_PASSWORD: ${{ secrets.pypi_token }}
+        run: pipx run twine upload artifacts/dist/*
+
+  make-container:
+    needs: [lint, wheel, test]
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Set outputs
+        id: vars
+        run: echo ::set-output name=repo_name::${GITHUB_REPOSITORY#*\/}
```

Comment:
What does this do?

Reply:
I was using it to set the builder argument for docker/build-push-action@v2 but I'm not sure what it is for. I've removed it and it still works.